addon/model-data.js

```diff
@@ -143,20 +159,31 @@ export default class M3ModelData {
     };
   }
 
+  _initBaseModelData(modelName, id) {
+    this.baseModelData = this.store.modelDataFor(modelName, id);
+    this.baseModelData._registerProjection(this);
+  }
+
+  _registerProjection(modelData) {
+    if (!this.__projections) {
+      // we ensure projections contains the base as well
+      // so we have complete list of all related model datas
+      this.__projections = [this];
+    }
+    this.__projections.push(modelData);
+  }
+
   /**
    *
    * @param updates
    * @param nestedCallback a callback for updating the data of a nested model-data instance
    * @returns {Array}
    * @private
    */
-  _mergeUpdates(updates, nestedCallback, calculateChanges = true) {
+  _mergeUpdates(updates, nestedCallback) {
     let data = this._data;
 
-    let changedKeys;
-    if (calculateChanges) {
-      changedKeys = [];
-    }
+    let changedKeys = [];
```

Comment:
what's the effect of no longer having calculateChanges? Do we overly notify or have we guarded against that by other means?

Reply:
The flag is more of an optimization - there is no need to calculate changes if we don't have a record to notify. In our case, however, the algorithm already walks the changes as it needs to execute special logic for applying them, therefore computing the changes have little overhead.